doc/man/man5/strinet.5.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) $Id$
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2009  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual
.\" under the terms of the GNU Free Documentation License, Version 1.3 or
.\" any later version published by the Free Software Foundation; with no
.\" Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.  A
.\" copy of the license is included in the section entitled "GNU Free
.\" Documentation License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaning to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF
.\" SUCH CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified $Date$ by $Author$
.\"
.\" -------------------------------------------------------------------------
.\"
.\" $Log$
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore T
.R2
.so openss7.macros
.\"
.\"
.TH @PACKAGE_UCNAME@ 5 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ System File Formats"
.SH NAME
.B @PACKAGE@
\- the @PACKAGE_TITLE@ package
.\"
.\"
.SH OVERVIEW
.PP
This is the
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@ - @VERSION@ )
package for
.BR "Linux Fast-STREAMS" .
.[
lfs
.]
.PP
OpenSS7
.I STREAMS
add-on packages are separate add-on packages to the
.B Linux Fast-STREAMS
.[
lfs
.]
release that provide additional functionality and protocol support not traditionally found on UNIX
systems, or which have traditionally fallen into the domain of value-add system software, or, in
this case, where the functionality was optional or differently handled in a number of implementations.  The
.B @PACKAGE_TITLE@
package is one of several optional add-on packages that provide enhanced
.I STREAMS
support beyond that provided by the base
.I STREAMS
package.  This package provides support for Internet Protocols based on the
.IR "Transport Provider Interface (TPI)" ,
.BR tpi (7),
not provided by the base
.I STREAMS
package.
.\"
.\"
.SH DESCRIPTION
.PP
The
.B @PACKAGE_TITLE@
package provides
.IR STREAMS (9)
modules and drivers, libraries, programs, initialization scripts, and daemons that
implement and provide
networking protocols based on the transport layer of the OSI
reference model.
.[
x200
.]
.PP
The package currently provides the following
.IR STREAMS (9)
modules and drivers:
.ad l
.IP \(en \w'\(en\(em'u
.B streams_inet.ko
driver
.PD 0
.IP \(en
.B streams_rawip.ko
driver
.IP \(en
.B streams_tcp.ko
driver
.IP \(en
.B streams_udp.ko
driver
.ad b
.PD
.PP
The following are
.I STREAMS
drivers installed by the
.B @PACKAGE@
package:
.TP
.BR inet (4)
OpenSS7 INET
.I STREAMS
driver.
The
.BR inet (4)
driver provides the first generation approach to INET
.I STREAMS
drivers: it uses an in-kernel
.I STREAMS
over
.I Sockets
approach.
.TP
.BR rawip (4)
Raw Internet Protocol
.I STREAMS
driver.
The
.BR rawip (4)
dirver provides a second generation raw IP driver that is implemented entirely
using
.IR STREAMS .
.TP
.BR tcp (4)
Transmission Control Protocol
.I STREAMS
driver.
The
.BR tcp (4)
dirver provides a second generation TCP driver that will be implemented entirely
using
.IR STREAMS .
.TP
.BR udp (4)
User Datagram Protocol
.I STREAMS
driver.
The
.BR udp (4)
dirver provides a second generation UDP driver that is implemented entirely
using
.IR STREAMS .
.PD
.PP
The
.B @PACKAGE_TITLE@
package includes kernel modules,
.I SVR 4.2
.[
svr42
.]
.I STREAMS
drivers, modules, libraries, utilities, test programs, daemons, and
development environment for the development and execution of
.B @PACKAGE_TITLE@
protocol stack components and applications for the
.B Linux Fast-STREAMS
environment.
.PP
This distribution is only currently applicable to
.B Linux
2.4 and 2.6 kernels and was targeted at
.IR ix86 ", " x86_64 ", " ppc " and " ppc64
architectures, but should build and install for other architectures as well.
.PP
This package is complete and has been in production use for
.I Linux Fast-STREAMS
for a number of years.
.\"
.\"
.SH "PACKAGE CONTENTS"
.PP
The
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@ - @VERSION@ )
package contains the following manual pages:
.\"
.HP
.BR strinet (5)\ \-\ 
@PACKAGE_TITLE@ package.
.PD 0
.HP
.BR strinet_mknod (8)\ \-\ 
create or remove
.I STREAMS
INET device nodes.
.PD
.\"
.\"
.HP
.BR xti_ip (3)\ \-\ 
X/Open Transport Interface - Corrigendum for Internet Protocol (IP) Conformance.
.PD 0
.HP
.BR xti_udp (3)\ \-\ 
X/Open Transport Interface - Corrigendum for User Datagram Protocol (UDP) Conformance.
.HP
.BR xti_tcp (3)\ \-\ 
X/Open Transport Interface - Corrigendum for Transmission Control Protocol (TCP) Conformance.
.HP
.BR t_kpalive (3)\ \-\ 
X/Open Transport Interface - Corrigendum for Transmission Control Protocol (TCP) Conformance.
.HP
.BR t_tcp_info (3)\ \-\ 
X/Open Transport Interface - Corrigendum for Transmission Control Protocol (TCP) Conformance.
.PD
.\"
.HP
.BR arp (4)\ \-\ 
User Datagram Protocol Stream.
.PD 0
.HP
.BR egp (4)\ \-\ 
Raw Internet Protocol Stream.
.HP
.BR ggp (4)\ \-\ 
Raw Internet Protocol Stream.
.HP
.BR icmp (4)\ \-\ 
Raw Internet Protocol Stream.
.HP
.BR idp (4)\ \-\ 
Raw Internet Protocol Stream.
.HP
.BR inet (4)\ \-\ 
OpenSS7 INET Devices.
.HP
.BR INET (4)\ \-\ 
OpenSS7 INET Devices.
.HP
.BR ip (4)\ \-\ 
Raw Internet Protocol Stream.
.HP
.BR ipip (4)\ \-\ 
Raw Internet Protocol Stream.
.HP
.BR pup (4)\ \-\ 
Raw Internet Protocol Stream.
.HP
.BR rawip (4)\ \-\ 
Raw Internet Protocol Stream.
.HP
.BR rawip2 (4)\ \-\ 
Raw Internet Protocol Stream.
.HP
.BR rip (4)\ \-\ 
User Datagram Protocol Stream.
.HP
.BR tcp (4)\ \-\ 
Transmission Control Protocol (TCP).
.HP
.BR tcp2 (4)\ \-\ 
Transmission Control Protocol (TCP).
.HP
.BR ticlts (4)\ \-\ 
connectionless loopback transport provider.
.HP
.BR ticots (4)\ \-\ 
connection oriented loopback provider.
.HP
.BR ticots_ord (4)\ \-\ 
connection oriented orderly release loopback provider.
.HP
.BR ticotsord (4)\ \-\ 
connection oriented orderly release loopback provider.
.HP
.BR udp (4)\ \-\ 
User Datagram Protocol Stream.
.HP
.BR udp2 (4)\ \-\ 
User Datagram Protocol Stream.
.PD
.\"
.HP
.BR tpi_ip (7)\ \-\ 
Transport Provider Interface (TPI) - Internet Protocol (IP) Corrigendum.
.PD 0
.HP
.BR tpi_udp (7)\ \-\ 
Transport Provider Interface (TPI) - User Datagram Protocol (UDP) Corrigendum.
.HP
.BR tpi_tcp (7)\ \-\ 
Transport Provider Interface (TPI) - Transmission Control Protocol (TCP) Corrigendum.
.PD
.\"
.SS "Test Programs"
.HP
.BR test-inet_raw (8)\ \-\ 
a test suite executable for the
.BR rawip (4)
.I STREAMS
driver.
.PD 0
.HP
.BR test-inet_tcp (8)\ \-\ 
a test suite executable for the
.BR tcp (4)
.I STREAMS
driver.
.HP
.BR test-inet_udp (8)\ \-\ 
a test suite executable for the
.BR udp (4)
.I STREAMS
driver.
.PD
.PP
In addition to these manual pages, the
.B @PACKAGE_TITLE@
package installs an
.IR "Installation and Reference Manual" .
.\"
.\"
.SH DEVICES
.PP
.TP
.BR inet (4)
first generation INET driver.
.TP
.BR rawip (4)
second generation RAW IP driver.
.TP
.BR tcp (4)
second generation TCP driver.
.TP
.BR udp (4)
second generation UDP driver.
.\"
.\"
.SH PREREQUISITES
.PP
The
.B @PACKAGE@
package requires the following packages to be installed:
.TP
.BR streams (5)
The
.B Linux Fast-STREAMS
base package.
.TP
.BR strcompat (5)
The
.B Linux Fast-STREAMS
compatibility modules package.
.TP
.BR strxns (5)
The
.B Linux Fast-STREAMS
X/Open Networking Services package.
.TP
.BR strxnet (5)
The
.B Linux Fast-STREAMS
XTI/TLI Library package.
.TP
.BR strnsl (5)
The
.B Linux Fast-STREAMS
Network Selection Library package.
.TP
.BR strsock (5)
The
.B Linux Fast-STREAMS
Sockets Library package.
.\"
.\"
.SH NOTICES
.PP
.I Production
releases are releases that have received exhaustive testing within the project
and validated on specific distributions and architectures.
.I Production
release packages do not ship with known defects.  Please rememaber that there
is
.B NO WARRANTY
on public release packages.
.PP
.I Production
packages ship containing a list of validated distributions and architectures.
Full regression testing of any maintenance changes is performed.  Suitability
of software in this category for production use on the specified target
distributions and architectures is at the discretion of the user.  It should
not be necessary to perform validation tests on the set of supported target
systems before considering production use.
.PP
.B @PACKAGE_TITLE@
could possibly contain unknown defects.  This is a
.I production
release.  Nevertheless, some remaining unknown defects could possibly be
harmful.  Validation testing has been performed by the
.I OpenSS7 Project
and external entities on this software for the set of systems listed in the
release notes.  Nevertheless, the software might still fail to configure or
compile on other systems.  The
.I OpenSS7 Project
recommends that you validate this software for your target system before using
this software.  Use at your own risk.  Remember that there is
.BR "NO WARRANTY" .
(See the section
.RI \(lq "NO WARRANTY" \(rq
under the
.RI \(lq "GNU General Public License" \(rq,
in the manual.)
.PP
This software is
.I production
software.  As such, it is stable on validated systems but might still crash
your kernel in unique circumstances.  Installation of the software on a
non-validated distribution might mangle your header files or Linux
distribution in such a way as to make it unusable and rebooting the system
might not repair the problem.  You possibly lose all the data on your system.
Because this software stands a chance of crashing your kernel, the resulting
unstable system could possibly destroy computer hardware or peripherals making
them unusable.  You might void the warranty on any system on which you run
this software.
.B YOU HAVE BEEN WARNED.
.\"
.\"
.SH BUGS
.PP
The
.B @PACKAGE@
package has been tested with a wide array of production
.B Linux
distributions, kernels, and target architectures, and is considered production grade by
.IR "The OpenSS7 Project" .
At the time of release, the
.B @PACKAGE@
package had no known bugs.  Problems can be reported to
.RB < bugs@openss7.org >.
Please, however, use the included
.B send-pr
script for generating actionable problem reports.
Please see the file
.RI \(lq PROBLEMS \(rq
in the release for problem reporting guidelines.
.\"
.\"
.SH "SEE ALSO"
.PP
.BR inet (4),
.BR tcp (4),
.BR udp (4),
.BR rawip (4).
.\"
.\"
.SH COMPATIBILITY
.PP
The
.B @PACKAGE@
package is compatible with
.I SVR 4.2
and systems based on
.I SVR 4.2
such as
.IR Solaris \(rg
and
.IR UnixWare \(rg,
with the following portability considerations:
.IP \(em \w'\(em\(em'u
.B @PACKAGE_TITLE@
provides compatibility with all implementations of the Internet Protocol suite
using the Transport Provider Interface Revision 1.5 or 2.0.0
specifications.
.[
tpi revision 2.0.0
.]
.IP \(em
These drivers are designed to operate with the XTI/TLI library implementation
provided by the
.BR strxnet (5)
package and is thus compatible with most XTI library implementations as well.
.\"
.\"
.SH CONFORMANCE
.PP
.IR TPI ,
.[
tpi revision 2.0.0
.]
.IR "XNS 5.2" ,
.[
xns 5.2
.]
.IR UNIX95 .
Conformance is tested using the
.BR test-inet_raw (8),
.BR test-inet_tcp "(8), and "
.BR test-inet_udp (8),
test case executables and the
.I @PACKAGE_TITLE@
.B autotest
test suite.
.\"
.\"
.SH HISTORY
.PP
The
.B @PACKAGE_TITLE@
package originally
contained the XTI/TLI library and
.BR timod (4)
and
.BR tirdwr (4)
modules.
The library and the
.BR timod (4)
and
.BR tirdwr (4)
modules were split from the
.BR strinet (5)
package and placed in their own
.BR strxnet (5)
package.  NSL library and Socket library functions were added to the
.BR strxnet (5)
package initially, but subsequently split to separate
.BR strsock (5)
and
.BR strnsl (5)
packages.
What remains in the
.BR strinet (5)
package is the header files, and
.I STREAMS
drivers, also TPI interface header files and documentation.
.PP
The
.B @PACKAGE_TITLE@
package is a stable, production grade release.
.\"
.\"
.SH "FUTURE DIRECTIONS"
.PP
The INET package is a stable production release and is largely in
maintenance mode.  There are, nevertheless, several items remaining to be
done.  See the
.RI \(lq TODO \(rq
file in the release for detailed information on further development.
.\"
.\"
.[
$LIST$
.]
.\"
.TI


NEWS
OpenSS7 Master Package -- history of user visible changes.  2009-04-04
$Id: NEWS,v 0.9.2.72 2009-04-19 00:28:25 brian Exp $
Copyright (c) 2008-2009  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
`OpenSS7 Master Package' package.

 Major changes for release openss7-0.9.2.F.20090405
---------------------------------------------------

This is the seventh `OpenSS7 Project' release of the `OpenSS7 Master
Package' package.  For a sub-package listing, see `README'
(http://www.openss7.org/codefiles/openss7/README) in the release.  Most
of the sub-packages in the release are production grade for `Linux
Fast-STREAMS'.(1)  All existing validation test suites run clean on
supported distributions, production kernels and architectures.

The original intention was that the `OpenSS7 Master Package' package
not be released as often as subpackages and, indeed, newer subpackages
can be strapped into the master package and built automatically.
However, in reality, the `OpenSS7 Master Package' package has been
released as often as the subpackages.  To rebuild the master package
with a new sub-package release, simply replace the directory to which
the sub-package belongs with the unpackged sub-package reelase and then
rebuild the master package.  This release provides support for recent
distributions, production kernels and tool chains.  The release also
provides support for significiant new features in the sub-packages.

Major features since the last public release are:

   - Addition of the `stratm' sub-package to the master build and
     release that now includes all of the ATM components that were
     previously part of the `strss7' sub-package.

   - Significant rework of the `strchan' sub-package into the `strss7',
     `strisdn', `strx25' and `striso' packages.  One single driver now
     meets the needs of all sub-packages.

   - Completion of the `strsock' package, moves from pre-alpha to beta.

   - The `strnsl' subpackage moves from alpha to production.

   - Significant completion of the `sigtran' package: all components
     moved back to production.


_This is a public, stable, production grade release of the package: it
deprecates previous releases.  Please upgrade to the current release
before reporting bugs._

As with other OpenSS7 releases, this release configures, compiles,
installs and builds RPMs and DEBs for a wide range of Linux 2.4 and 2.6
`rpm(8)' and `dpkg(1)' based distributions, and can be used on
production kernels without patching or recompiling the kernel.

This package is publicly released strictly under the `GNU Affero
General Public License Version 3'; however, commercial licensing is
available.  The release is available as an `autoconf(1)' tarball, SRPM,
DSC, and a set of binary RPMs and DEBs for popular distributions.  See
the package page (http://www.openss.org/openss7_pkg.html) for the
`autoconf(1)' tarballs, SRPMs, DSCs and binary RPMs and DEBs.  See the
repository page (http://www.openss7.org/repos.html) for information
concerning network installation and update sources.

See
`http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/ChangeLog'
and `http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/NEWS' in
the release for more information.  Also, see the `openss7.pdf' manual
in the release (also in html
(http://www.openss7.org/openss7_manual.html)).

For the news release, see `http://www.openss7.org/rel20081029_L.html'.

---------- Footnotes ----------

(1) Sub-packages are no longer supported on the deprecated `LiS' STREAMS
package.

 Major changes for release openss7-0.9.2.G
------------------------------------------

This is the sixth public release of the OpenSS7 Master Package.  See
`README' in the release for a sub-package listing.  Most of the
sub-packages in the release are production grade for `Linux
Fast-STREAMS'.  All existing validation test suites run clean on
supported distributions and architectures.

The `OpenSS7 Master Package' is not released as often as the
sub-packages.  As sub-packages are released more often, to rebuild the
master package with a new sub-package release, simply replace the
directory to which the sub-package belongs with the unpacked
sub-package release and then rebuild the master package.  This release
provides support for recent distributions and tool chains.

Major features since the last public release are as follows:

   - License upgrade to `AGPL Version 3'.

   - Modifications to build under `Fedora' `2.6.22.5-49' kernel.  These
     changes also support `2.6.22.9-91.fc7' kernel.  Modifications to
     build under `Fedora' `2.6.25-45.fc9' and `2.6.26.5-45.fc9' kernels.
     Documented `lib32gcc1' problem on Ubuntu.  Noted problem running
     under XEN kernels.  XEN kernels are not yet supported.  Added
     `MODULE_VERSION' to all modules and drivers.

   - Ability to strap out major documentation build and installation
     primarily for embedded targets.  Improvements to common build
     process for embedded and cross-compile targets.  Cross-compile
     fixes (strap out `AC_FUNC_REALLOC' macro when cross-compiling).
     Conversion of RPM spec files to common approach for major
     subpackages.

     Build system now builds `yum(8)' repositories for RPMs and
     `apt-get(8)' repositories for DEBs.  Installation documentation
     has been updated to include details of repository install
     sourcesref.

   - Higher performance and updated performance papers.

   - Updated tool chain to `m4-1.4.12', `autoconf-2.63' and
     `texinfo-4.13'.  Support for `flex 2.5.33' in maintainer mode.

   - Updated references database for manual pages and roff documents.

   - Added the following major sub-packages to the master build (and
     release):


src/util/init/strinet.sh
#!/bin/sh
#
# @(#) $RCSfile$ $Name$($Revision$) $Date$
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
# All Rights Reserved.
#
# Distributed by OpenSS7 Corporation.  See the bottom of this script for copying
# permissions.
#
# These are arguments to update-rc.d ala chkconfig and lsb.  They are recognized
# by openss7 install_initd and remove_initd scripts.  Each line specifies
# arguments to add and remove links after the the name argument:
#
# strinet:	start and stop strinet subsystem
# update-rc.d:	start 33 S . stop 33 0 6 .
# config:	/etc/default/strinet
# probe:	false
# hide:		false
# license:	GPL
# description:	This STREAMS init script is part of Linux Fast-STREAMS.  \
#		It is responsible for ensuring that the necessary STREAMS \
#		character devices are present in the /dev directory and \
#		that the STREAMS INET subsystem is configured and loaded.
#
# LSB init script conventions
#
### BEGIN INIT INFO
# Provides: strinet
# Required-Start: streams $network
# Required-Stop: streams $network
# Default-Start: 3 4 5
# Default-Stop: 0 1 2 6
# X-UnitedLinux-Default-Enabled: yes
# Short-Description: start and stop STREAMS INET subsystem
# License: GPL
# Description:	This STREAMS INET init script is part of Linux Fast-STREAMS.
#	It is reponsible for ensuring that the necessary STREAMS INET character
#	devices are present in the /dev directory and that the STREAMS INET
#	subsystem is configured and loaded.
### END INIT INFO

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
name='strinet'
config="/etc/default/$name"
desc="the STREAMS INET subsystem"
mknod="${name}_mknod"

[ -e /proc/modules ] || exit 0

if test -z "$STRINET_MKNOD" ; then
    for STRINET_MKNOD in /sbin/${name}_mknod /usr/sbin/${name}_mknod /bin/${name}_mknod /usr/bin/${name}_mknod ; do
	if [ -x $STRINET_MKNOD ] ; then
	    break
	else
	    STRINET_MKNOD=
	fi
    done
fi

if test -z "$INET_MKDEV" ; then
    for INET_MKDEV in /sbin/inet_mkdev /usr/sbin/inet_mkdev /bin/inet_mkdev /usr/bin/inet_mkdev ; do
	if [ -x $INET_MKDEV ] ; then
	    break
	else
	    INET_MKDEV=
	fi
    done
fi

if test -z "$INET_RMDEV" ; then
    for INET_RMDEV in /sbin/inet_rmdev /usr/sbin/inet_rmdev /bin/inet_rmdev /usr/bin/inet_rmdev ; do
	if [ -x $INET_RMDEV ] ; then
	    break
	else
	    INET_RMDEV=
	fi
    done
fi

# Specify defaults

[ -n "$STRINET_PRELOAD"       ] || STRINET_PRELOAD=""
[ -n "$STRINET_DRIVERS"       ] || STRINET_DRIVERS="streams-inet streams-rawip streams-udp"
[ -n "$STRINET_MODULES"       ] || STRINET_MODULES=""
[ -n "$STRINET_MAKEDEVICES"   ] || STRINET_MAKEDEVICES="yes"
[ -n "$STRINET_REMOVEDEVICES" ] || STRINET_REMOVEDEVICES="yes"

# Source config file
for file in $config ; do
    [ -f $file ] && . $file
done

[ -z "$STRINET_MKNOD" -a -z "$INET_MKDEV" ] && STRINET_MAKEDEVICES="no"
[ -z "$STRINET_MKNOD" -a -z "$INET_RMDEV" ] && STRINET_REMOVEDEVICES="no"

RETVAL=0

umask 077

case :$VERBOSE in
    :no|:NO|:false|:FALSE|:0|:)
	redir='>/dev/null 2>&1'
	;;
    *)
	redir=
	;;
esac

build_options() {
    # Build up the options string
    :
}

start() {
    echo -n "Loading STREAMS kernel modules: "
    RETVAL=0
    modules=
    for module in $STRINET_PRELOAD ; do
	modules="${modules:+$modules }$module"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if ! eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -k -q -- $module $redir"
	    [ $? -eq 0 ] || echo -n "(failed)"
	fi
    done
    echo "."

    echo -n "Starting $desc: $name "
    build_options
    if [ $? -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
	RETVAL=1
    fi

    if eval "grep '^[[:space:]]*${name}[/.]' /etc/sysctl.conf $redir" ; then
	echo -n "Reconfiguring kernel parameters: "
	eval "sysctl -p /etc/sysctl.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi

    if [ -f /etc/${name}.conf ] ; then
	echo -n "Configuring STREAMS parameters: "
	eval "sysctl -p /etc/${name}.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi

    if [ -n "$STRINET_MKNOD" -o -n "$INET_MKDEV" ] ; then
	if [ :"$STRINET_MAKEDEVICES" = ":yes" ] ; then
	    if [ -n "$INET_MKDEV" ] ; then
		echo -n "Making STREAMS iBCS devices: "
		$INET_MKDEV
		if [ $? -eq 0 ] ; then
		    echo "."
		else
		    echo "(failed.)"
		    RETVAL=1
		fi
	    fi
	    if [ -n "$STRINET_MKNOD" ] ; then
		echo -n "Making STREAMS INET devices: "
		$STRINET_MKNOD
		if [ $? -eq 0 ] ; then
		    echo "."
		else
		    echo "(failed.)"
		    RETVAL=1
		fi
	    fi
	fi
    fi
    return $RETVAL
}

stop() {
    echo "Stopping $desc: $name "
    RETVAL=0
    if [ -n "$STRINET_MKNOD" -a ":$STRINET_REMOVEDEVICES" = ":yes" ] ; then
	echo -n "Removing STREAMS devices: "
	$STRINET_MKNOD --remove
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    echo -n "Unloading STREAMS kernel modules: "
    modules=
    for module in $STRINET_PRELOAD $STRINET_DRIVERS $STRINET_MODULES ; do
	modules="$module${modules:+ $modules}"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -r -q -- $module $redir"
	    if [ $? -ne 0 ] ; then
		echo -n "(failed) "
		RETVAL=1
	    fi
	fi
    done
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

restart() {
    stop
    start
    return $?
}

show() {
    echo "$name.sh: show: not yet implemented." >&2
    return 1
}

usage() {
    echo "Usage: /etc/init.d/$name.sh (start|stop|restart|force-reload|show)" >&2
    return 1
}

case "$1" in
    (start|stop|restart|show)
	$1 || RETVAL=$?
	;;
    (force-reload)
	restart || RETVAL=$?
	;;
    (*)
	usage || RETVAL=$?
	;;
esac

[ "${0##*/}" = "$name.sh" ] && exit $RETVAL

# =============================================================================
# 
# @(#) $RCSfile$ $Name$($Revision$) $Date$
#
# -----------------------------------------------------------------------------
#
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
#
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU Affero General Public License as published by the Free
# Software Foundation; version 3 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
# details.
#
# You should have received a copy of the GNU Affero General Public License along
# with this program.  If not, see <http://www.gnu.org/licenses/>, or write to
# the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
#
# -----------------------------------------------------------------------------
#
# U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
# behalf of the U.S. Government ("Government"), the following provisions apply
# to you.  If the Software is supplied by the Department of Defense ("DoD"), it
# is classified as "Commercial Computer Software" under paragraph 252.227-7014
# of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
# successor regulations) and the Government is acquiring only the license rights
# granted herein (the license rights customarily provided to non-Government
# users).  If the Software is supplied to any unit or agency of the Government
# other than DoD, it is classified as "Restricted Computer Software" and the
# Government's rights in the Software are defined in paragraph 52.227-19 of the
# Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
# the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
# (or any successor regulations).
#
# -----------------------------------------------------------------------------
#
# Commercial licensing and support of this software is available from OpenSS7
# Corporation at a fee.  See http://www.openss7.com/
#
# -----------------------------------------------------------------------------
#
# Last Modified $Date$ by $Author$
#
# -----------------------------------------------------------------------------
#
# $Log$
# Revision 0.9.2.18  2008-10-21 22:42:16  brian
# - handle verbose better in debian initscripts
#
# Revision 0.9.2.17  2008-04-28 22:52:17  brian
# - updated headers for release
#
# Revision 0.9.2.16  2007/08/14 04:28:49  brian
# - GPLv3 header update
#
# =============================================================================
# vim: ft=sh sw=4 tw=80
